### DISCUSSION TITLE
Wreck Monsters Delete Corpses & IDs

### DISCUSSION THREAD

**Discussion by VaelophisNyx** [FakeFishGames/Barotrauma#12410] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

So this is an issue I've seen a few times when wreck salvaging; a wreck will need an ID to open specific lockers, and sometimes that wreck will contain a monster that proceeds to eat all corpses within reach when within AI range. This can, if you don't rush to the wreck on the chance it's happening, lead to most if not all containers in the wreck being unopenable.

I believe this could be fixed by simply spawning a body bag when a human corpse is eaten, wreck-body or otherwise. The similar issue of a player corpse, or bot, being eaten by crawlers/mudraptors and completely erased along with most of their (usually expensive) gear is quite frustrating.

### Reproduction steps

1. Start a campaign or wreck salvage mission
2. approach a wreck
3. if the wreck contains aliens, observe that they usually consume any corpses in the room with them
4. depending on the wreck (potentially only custom wrecks?) you may potentially be locked out of numerous containers you otherwise should have access to with a shipwreck ID

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This is not really a bug, I'll convert this to a discussion

**Comment by G-M-twostay** [comment]
i kind of feel like this incentivize us to rush to wreck. It makes sense that sometimes there will be nothing remaining in the wreck as monsters ate them all. It also makes sense that if you salavage a wreck earlier the chances of finding full body is greater. Tho I do wish a way of opening secured cabinet without id card; however, to prevent greifing, this way might be very timely or loud so that doing it secretly wouldn't be very possible.

**Comment by Hazmat616** [comment]
Have corpses eaten become divers remains as guaranteed loot that will be dismantled into some of their items and a chance for their id.

**Reply by VaelophisNyx** [reply]
I'd say guarantee for the ID and chance for other loot rather, otherwise that's just this issue with extra RNG
